proposals/0245-sharing-wifi-ssid-and-password.md

```diff
@@ -7,112 +7,236 @@
 
-### Changes for communicating Credentials from HMI to Core
+### HMI API changes
 
-HMI will create a WiFi access point and communicate these using below HMI API.
+1. HMI will create a WiFi access point and communicate these using below HMI API.
+2. HMI will notify SDL Core using `OnWiFiTransportStatusUpdate` whenever WiFi state is changed.
+3. Device Info is changed to provide WiFi auto-connect feature support information to HMI. 
+4. HMI will receive `DeviceInfo.supportWiFiAutoConnect` in `RegisterAppInterface` Request using `primary transport`.
+5. HMI can send `OnSystemRequest` notification with `RequestType` as `CONNECT_WIFI` to inform app to try connecting WiFi. This notification will be sent only when a user has provided consent to share WiFi credentials, and the connected app supports WiFi auto-connect and the user attempts to launch that app.
+6. HMI will not send `OnSystemRequest` notification with `RequestType` as `CONNECT_WIFI` if app is already connected using WiFi.
 
 ```
-<interface name="BasicCommunication" version="2.1.0" date="2019-XX-XX">
+<interface name="Common" version="X.X.X" date="2019-XX-XX">
+<struct name="DeviceInfo" since="X.X">
+  .
+  .
+  .
+  <param name="supportWiFiAutoConnect" type="Boolean" mandatory="false">
+    <description>support to connect WiFi enabled by developer and is supported by mobile device.</description>
+  </param>         
+</struct> 
+
+<enum name="RequestType">
+  .
+  .
+  .
+  <element name="CONNECT_WIFI" />
+</enum>
+</interface>
+
+<interface name="BasicCommunication" version="X.X.X" date="2019-XX-XX">
 	.
 	.
 	.
-  <function name="OnWiFiTransportAvailable" messagetype="notification">
+  <function name="OnWiFiTransportStatusUpdate" messagetype="notification">
       <description>HMI must notify SDL about WiFi Access point.</description>
-	    <param name="ssid" type="String" minlength="1" maxlength="30" mandatory="true">
+	    <param name="ssid" type="String" minlength="1" maxlength="32" mandatory="false">
         <description>name of the WiFi ssid</description>
       </param>
-	    <param name="password" type="String" minlength="1" maxlength="30" mandatory="true">
+	    <param name="password" type="String" minlength="1" maxlength="100" mandatory="false">
         <description>password to use to connect AP</description>
       </param>
-	    <param name="securityType" type="WiFiSecurityType" mandatory="true">
+	    <param name="securityType" type="WiFiSecurityType" mandatory="false">
         <description>security type of WiFi AP</description>
       </param>
+      <param name="wifiState" type="WiFiStateInfo" mandatory="false">
+        <description>wifi state info</description>
+      </param>
   </function>
 </interface>
 ```
 
-Enum to define WiFi security type.
+Enum to define WiFi state and WiFi security type.
 
 ```

```

Comment:
Same comment as above regarding code block formatting here.

Reply:
Added changes in [this ](https://github.com/smartdevicelink/sdl_evolution/pull/819/commits/8cecbd20a37e7db3d90e66bb3a338fabd1aa3c93)commit.